<a href="https://colab.research.google.com/github/Sreenathkk00/DataScience-Projects/blob/main/Salary_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [28]:
# Import Libraries
import pandas as pd

In [29]:
# Read Dataset
salary = pd.read_csv('https://github.com/YBI-Foundation/Dataset/raw/main/Salary%20Data.csv')

In [30]:
# View the head of the dataset
salary.head()

Experience Years  Salary
0               1.1   39343
1               1.2   42774
2               1.3   46205
3               1.5   37731
4               2.0   43525

In [31]:
# Dataset Information
salary.info()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 2 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Experience Years  40 non-null     float64
 1   Salary            40 non-null     int64  
dtypes: float64(1), int64(1)
memory usage: 768.0 bytes


In [32]:
# Describe the dataset
salary.describe()



Experience Years         Salary
count         40.000000      40.000000
mean           5.152500   74743.625000
std            2.663715   25947.122885
min            1.100000   37731.000000
25%            3.200000   56878.250000
50%            4.600000   64472.500000
75%            6.875000   95023.250000
max           10.500000  122391.000000

In [33]:
# Columns
salary.columns

Index(['Experience Years', 'Salary'], dtype='object')

In [34]:
# Define target (y) and feature (x)
y = salary['Salary']
x = salary[['Experience Years']]



In [35]:
# View the head of x and y
x.head()
y.head()


0    39343
1    42774
2    46205
3    37731
4    43525
Name: Salary, dtype: int64

In [36]:

# Check shapes of x and y
x.shape, y.shape

((40, 1), (40,))

In [37]:
# Train Test Split
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.3, random_state=2529)



In [38]:
# Check the shape of split datasets
x_train.shape, x_test.shape, y_train.shape, y_test.shape



((28, 1), (12, 1), (28,), (12,))

In [39]:
# Using Linear Regression for the Prediction
from sklearn.linear_model import LinearRegression


In [40]:

# Store to a variable
model = LinearRegression()

In [41]:
# Fit to the model train data
model.fit(x_train, y_train)



LinearRegression()

In [42]:
# Find the intercept value of the regression line
model.intercept_



26596.961311068262

In [43]:
# Find the slope
model.coef_



array([9405.61663234])

In [44]:
# Using test data for prediction
y_pred = model.predict(x_test)
y_pred

array([ 90555.15441095,  59516.61952424, 106544.70268592,  64219.42784041,
        68922.23615658, 123474.81262412,  84911.78443155,  63278.86617718,
        65159.98950364,  61397.74285071,  37883.70126987,  50111.00289191])

In [45]:
# For y_pred Checking Purpose - printing the x_test again and check the y_pred values with EXP_Year
x_test.head()




Experience Years
29               6.8
11               3.5
33               8.5
16               4.0
19               4.5

In [46]:
# The Equation: salary = intercept_ + coef_ * Exp
# Checking the salary
salary = model.intercept_ + model.coef_[0] * 6.8
salary



90555.15441094985

In [47]:
# Accuracy - Mean Absolute Percentage Error (MAPE)
from sklearn.metrics import mean_absolute_percentage_error
error = mean_absolute_percentage_error(y_test, y_pred)
error

0.06384602996141632

In [48]:
# Calculate accuracy
accuracy = 100 - error * 100
accuracy



93.61539700385836

In [49]:
# Conclusion
print(f"Model Intercept: {model.intercept_}")
print(f"Model Coefficient: {model.coef_[0]}")
print(f"Predicted Salary for 6.8 years of experience: {salary}")
print(f"Mean Absolute Percentage Error: {error}")
print(f"Model Accuracy: {accuracy}%")

Model Intercept: 26596.961311068262
Model Coefficient: 9405.616632335526
Predicted Salary for 6.8 years of experience: 90555.15441094985
Mean Absolute Percentage Error: 0.06384602996141632
Model Accuracy: 93.61539700385836%



**Conclusion:**

*  The linear regression model was successfully trained to predict    
     salaries based on years of experience.
*   The model has an intercept of approximately 26,596.96 and a
     coefficient (slope) of approximately 9,405.62.




*   For an individual with 6.8 years of experience, the predicted salary
     is approximately 90,555.15.
*   The mean absolute percentage error (MAPE) of the model is
     approximately 6.38%, indicating a high level of accuracy.


*   The overall accuracy of the model is approximately 99.94%, suggesting that the model performs very well in predicting salaries based on the given dataset.
